thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/model/merge/AverageAnomalyMergeModel.java

```diff
@@ -28,18 +28,26 @@ public void update(AnomalyDetectionContext anomalyDetectionContext,
 
     double weight = 0d;
     double score = 0d;
+    double baseline = 0d;
+    double current = 0d;
     for (RawAnomalyResultDTO rawAnomaly : rawAnomalyResultDTOs) {
       weight += rawAnomaly.getWeight();
       score += rawAnomaly.getScore();
+      current += rawAnomaly.getAvgCurrentVal();
+      baseline += rawAnomaly.getAvgBaselineVal();
     }
     if (rawAnomalyResultDTOs.size() != 0) {
       double size = rawAnomalyResultDTOs.size();
       weight /= size;
       score /= size;
+      current /= size;
```

Comment:
there computation is not correct (not consistent with raw weight, score, current, baseline definition) if you have different lengths for two anomalies to merge.